community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactions.java

```diff
@@ -249,4 +250,18 @@ private void assertDatabaseIsRunning()
             throw new DatabaseShutdownException();
         }
     }
+
+    public Snapshot snapshot()
+    {
+        return new KernelTransactionsSnapshot( allTransactions );
+    }
+
+    public interface Snapshot extends BooleanSupplier
```

Comment:
Maybe this interface should not extend `BooleanSupplier` and than `getAsBoolean()` could get a more meaningful name like `allTransactionsClosed()`